src/ophyd_async/epics/areadetector/drivers/aravis_driver.py

```diff
@@ -0,0 +1,157 @@
+                "YUV411Packed",
+            ),
+            1304e-6: ("Mono12", "BayerRG12", "YUV422Packed"),
+            1961e-6: ("RGB8Packed", "BGR8Packed", "YUV444Packed"),
+        }
+    ),
+    # cite: https://cdn.alliedvision.com/fileadmin/content/documents/products/cameras/various/appnote/GigE/GigE-Cameras_AppNote_PIV-Min-Time-Between-Exposures.pdf retrieved 2024-04-05  # noqa: E501
+    "Manta G-609": lambda _: 47e-6,
+    # cite: https://cdn.alliedvision.com/fileadmin/content/documents/products/cameras/Mako/techman/Mako_TechMan_en.pdf retrieved 2024-04-05  # noqa: E501
+    "Mako G-040": _reverse_lookup(
+        {
+            101e-6: (
+                "Mono8",
+                "Mono12Packed",
+                "BayerRG8",
+                "BayerRG12Packed",
+                "YUV411Packed",
+            ),
+            140e-6: ("Mono12", "BayerRG12", "YUV422Packed"),
+            217e-6: ("RGB8Packed", "BGR8Packed", "YUV444Packed"),
+        }
+    ),
+    "Mako G-125": lambda _: 70e-6,
+    # Assume 12 bits: 10 bits = 275e-6
+    "Mako G-234": _reverse_lookup(
+        {
+            356e-6: (
+                "Mono8",
+                "BayerRG8",
+                "BayerRG12",
+                "BayerRG12Packed",
+                "YUV411Packed",
+                "YUV422Packed",
+            ),
+            # Assume 12 bits: 10 bits = 563e-6
+            726e-6: ("RGB8Packed", "BRG8Packed", "YUV444Packed"),
+        }
+    ),
+    "Mako G-507": _reverse_lookup(
+        {
+            270e-6: (
+                "Mono8",
+                "Mono12Packed",
+                "BayerRG8",
+                "BayerRG12Packed",
+                "YUV411Packed",
+            ),
+            363e-6: ("Mono12", "BayerRG12", "YUV422Packed"),
+            554e-6: ("RGB8Packed", "BGR8Packed", "YUV444Packed"),
+        }
+    ),
+}
+
+
+class AravisDriver(ADBase):
+    # If instantiating a new instance, ensure it is supported in the _deadtimes dict
+    """Generic Driver supporting the Manta and Mako drivers.
+    Fetches deadtime prior to use in a Streaming scan.
+    Requires driver firmware up to date:
+    - Model_RBV must be of the form "^(Mako|Manta) (model)$"
+    """
+
+    def __init__(self, prefix: str, name: str = "") -> None:
+        self.trigger_mode = ad_rw(AravisTriggerMode, prefix + "TriggerMode")
+        self.trigger_source = ad_rw(AravisTriggerSource, prefix + "TriggerSource")
+        self.model = ad_r(str, prefix + "Model")
+        self.pixel_format = ad_rw(str, prefix + "PixelFormat")
+        self.dead_time: Optional[float] = None
+        super().__init__(prefix, name=name)
+
+    async def _fetch_deadtime(self) -> None:
+        # All known in-use version B/C have same deadtime as non-B/C
+        model: str = (await self.model.get_value()).removesuffix("B").removesuffix("C")
+        pixel_format: str = await self.pixel_format.get_value()
+        self.dead_time = _deadtimes.get(model, lambda _: nan)(pixel_format)
```

Comment:
As above, suspicious about use of `NaN` here, think I would rather it causes an immediate descriptive error.